conans/client/generators/pkg_config.py

```diff
@@ -83,7 +83,10 @@ def single_pc_file_contents(self, name, cpp_info):
              ["-D%s" % d for d in cpp_info.defines]]))
 
         if cpp_info.public_deps:
-            public_deps = " ".join(cpp_info.public_deps)
+            pkg_config_names = []
+            for public_dep in cpp_info.public_deps:
+                pkg_config_names += [self.deps_build_info[public_dep].get_name("pkg_config")]
+            public_deps = " ".join(pkg_config_names)
```

Comment:
I think that this will break previous behavior.

Reply:
The `cpp_info.public_deps` must contain the name of the reference, not the name given in the `cpp_info.name` or `cpp_info.names["generator"]`

Reply:
yes, I mean public_deps in this function, the one used to generate "Requires" field of pkg-config file, this one should contain name given in the cpp_info.name or cpp_info.names["generator"].

Reply:
Yes, I think you are right. It's not the public_deps but the `cpp_info.name/s` of that deps 👍